docs/proposals/config.md

```diff
@@ -0,0 +1,109 @@
+    --grpc-address     "0.0.0.0:9091" \
+    --cluster.peers    "thanos-cluster.example.org" \
+    --config           "promA.yaml;promB.yaml" \
+```
+
+The configuration supplied to the config component will then be distributed to the peers that config is aware of and exposed via a gRPC API so that each sidecar can get its configuarion.
+The gRPC endponit will take a request from a sidecar and based on the sidecar making the request it will give back the computed configuration.
+
+`thanos config` will initially only support a subset of the Prometheus Config will initially include a subset of global and scrape_config along with static_sd_config and file_sd_config.
+
+The inital configuration strategy will be intended for a multi-tenant use case and so keep jobs with a given label together on a single Prometheus instance.
+
+### Thanos Sidecar
+
+With the addition of `thanos config` we will look to update the `thanos sidecar` to periodically get its configuration from the config component. Each time it does so it will update the Prometheus instance by either updating the targets via file_sd_config or adding a new configuration job and forcing a reload on the Prometheus configuration.
+
+To do this the sidecar will need to know the location of the config component to get its configuration.
+
+```
+$ thanos sidecar \
+    --tsdb.path        "/path/to/prometheus/data/dir" \
+    --prometheus.url   "http://localhost:9090" \
+    --gcs.bucket       "example-bucket" \
+    --cluster.peers    "thanos-cluster.example.org" \
+    --config.url       "thanos-cluster.config.org" \
+```
+
+### Client/Server Backwards/Forwards compatibility
+
+This change would be fully backwards compatible, you do not have to use the config component within your Thanos setup and can still choose to manually set up configuration and service discovery for each node.
+
+The alternatives below might be a good starting point for users that do not need to worry about re-allocation of targets when the number of Prometheus instances in a cluster scales.
+
+## Alternatives considered
+
+### Prometheus & Hashmod
+
+An alternative to this is to use the existing [hashmod](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config) functionality within Prometheus to enable [horizontal scaling](https://www.robustperception.io/scaling-and-federating-prometheus/), in this scenario a user would supply their entire configuaration to every Prometheus node and use hashing to scale out.
+The main downside of this is that as a Prometheus instance is added or removed from the cluster the targets will be moved to a new scrape instance. This is bad if you are wanting to keep tenants to a minimal number of Prometheus instances.
+We have seen this approach also have a problem with hot shards in the past whereby a given Prometheus instance may endup being overloaded and is difficult to redistribute the load onto other Promethus instances in the scrape pool.
```

Comment:
this very strong argument, and should be placed IMO in the very up in `motivation` section